### PR TITLE
add custom address function

### PR DESCRIPTION
**Pull request by dcousens**
I think this is pretty damn neat :+1: 

Prepares the ecosystem for `4.0.0` :)

### CODE REVIEW

File: test/fixtures/chain.json [hunk 1]
```diff
@@ -34,5 +51,22 @@
       "1BJ2UQpsyRgo8ZNaVPFBtvtRX2F9tCVtan": 10
     },
     "node": "xprv9s21ZrQH143K2muoCkQq38JjsN4PGCovkjh7LVauc5cgRnkVoEtt1cDA5AjkFTBwfMwJrqw2JRWZk7YkvEQDFScF1BNJoz7CaPeqjkU1WYW"
+  },
+  {
+    "addressFunction": "eth",
```
Comment (dcousens) [inline test/fixtures/chain.json]: an notable side-effect of how this could work

Comment (dcousens) [reply test/fixtures/chain.json]: uh shit,  its meant to be `sha3` ... moment

File: test/chain.js [hunk 2]
```diff
@@ -2,28 +2,47 @@ var bitcoin = require('bitcoinjs-lib')
...
+  var hash = bitcoin.crypto.hash160(node.getPublicKeyBuffer())
+  var script = bitcoin.script.witnessPubKeyHash.output.encode(hash)
+  return bitcoin.address.fromOutputScript(script)
+}
+
+function ethAddr (node) {
+  return '0x' + createKeccakHash('keccak256')
+    .update(node.getPublicKeyBuffer())
+    .digest().toString('hex')
+}
```
Comment (dcousens) [inline test/chain.js]: A notable side-effect of how this could work

Comment (fanatid) [reply test/chain.js]: @dcousens I do not understand, why not `digest('hex')`?

**Comment by fanatid** [comment]
lgtm

File: test/chain.js (hunk 2 above)
Comment (dcousens) [reply test/chain.js]: > digest('hex')

Forgot that was an option - any advantage?

**Comment by dcousens** [comment]
thanks for fixing the tests @fanatid

**Pull request merged**